app/models/projects/exports/pdf_export/info_map.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Projects::Exports::PDFExport
+  module InfoMap
+    def build_flat_meta_infos_map(projects)
+      infos_map = {}
+      projects.each_with_index do |project, index|
+        infos_map[project.id] = { level_path: [index + 1], level: 0, children: [], project: }
+      end
```

Comment:
This portion could be simplified to:

```suggestion
      infos_map = projects.map.with_index do |project, index|
        [project.id, level_path: [index + 1], level: 0, children: [], project:]
      end.to_h
```

Reply:
Much nicer! 👍